IPv6 support for iBFT iSCSI boot

yukariatlas opened the pull request:
Use IPv4-specific APIs during NIC setup based on iBFT information, regardless of the IP version, which leads to the following error in IPv6-only environments:

Setting up software interface ens300f0np0
iscsistart: Invalid or missing ipaddr in fw entry
Setting up software interface ens300f0np0 failed

To address this, the process now determines the IP version first and calls the appropriate functions to handle NIC setup.

gonzoleeman reviewed (changes requested):
Thank you for this submission! I have a few small comments, if you wouldn't mind addressing them.

yukariatlas commented on usr/iscsi_net_util.c:
```diff
@@ -234,8 +236,57 @@ static char *find_vlan_dev(char *netdev, int vlan_id) {
+				  physdev, errno, strerror(errno));
+			return errno;
+		}
+	}
```
The ifr_name differs (physdev vs. netdev), with physdev existing only in VLAN scenarios.

gonzoleeman replied:
So it's setting up the VLAN, if needed, in addition to the regular interface? Either way, this was existing code.

yukariatlas commented on usr/iscsi_net_util.c:
```diff
@@ -409,6 +440,127 @@ int net_setup_netdev(char *netdev, char *local_ip, char *mask, char *gateway,
+	if (vlan_id && !netdev) {
+		log_error("No matching VLAN found for fw entry.\n");
+		return EINVAL;
+	}
```
Thank you for your feedback. The logic aligns with the net_setup_netdev_ipv4 function, but your concern is valid. I propose introducing two new variables: vlandev and targetdev. The targetdev variable will initially point to netdev and switch to vlandev in VLAN scenarios. Ultimately, we can check the existence of vlandev instead of netdev. If you find this approach suitable, I will update the logic in both the net_setup_netdev_ipv4 and net_setup_netdev_ipv6 functions. Also, would you be okay with me combining these changes into a single commit?

gonzoleeman replied:
I see you have done a separate commit to fix the variable overloading issue. Thank you.

yukariatlas commented on usr/iscsi_net_util.c:
```diff
@@ -409,6 +440,127 @@ int net_setup_netdev(char *netdev, char *local_ip, char *mask, char *gateway,
+	if (sock >= 0)
+		close(sock);
+	if (vlan_id)
+		free(netdev);
```
Building on the previous proposal, we can directly check vlandev instead of mixing its usage with netdev.

gonzoleeman replied:
Again, I see you've handled it, so no problem.

yukariatlas commented:
@gonzoleeman I have addressed all your suggestions by introducing an additional patch to clearly separate the usage. Please take a look and feel free to share any further feedback. Thank you!

gonzoleeman commented:
> @gonzoleeman I have addressed all your suggestions by introducing an additional patch to clearly separate the usage. Please take a look and feel free to share any further feedback. Thank you!

I will check it out. Thank you for your effort!

gonzoleeman commented:
Thank you for your submission, and for your patience.
